packages/app/src/cli/utilities/developer-platform-client/app-management-client.ts

```diff
@@ -471,9 +471,9 @@ export class AppManagementClient implements DeveloperPlatformClient {
           message: '',
           appModuleVersions: versionInfo.appModules.map((mod: ReleasedAppModuleFragment) => {
```

Comment:
Doing a side-by-side comparison, I see that:

1. `appVersionByTag` converts the module's config to a string, `activeAppVersion` does not
2. `appVersionByTag` uses the specification's `externalIdentifier` while `activeAppVersion` uses the `identifier`.

Otherwise they are identical.

I need to investigate which is the correct behavior, or if there's a legitimate reason to distinguish the cases. In which case, that should be recorded somewhere.

Reply:
RE config conversion to a string, looks like we rely on it being a string here, to match how it works in Partners:

https://github.com/Shopify/cli/blob/8d2d454b8263e3a17379070d8d810b399e8dce62/packages/app/src/cli/services/release.ts#L45

But presumably `activeAppVersion` leaving as an object is intentional (a bit harder to research since it's used in so many code paths).

I think leaving as an object is correct, and the one place that depends on it being a String (and immediately parses it to an object!) should be adjusted to account for non-strings as well.

RE `identifier` vs `externalIdentifier` it seems like `identifier` is the correct option, I can't see any situation where `externalIdentifier` would be correct.